cli/args/mod.rs

```diff
@@ -474,15 +562,18 @@ pub struct CliOptions {
   // application need not concern itself with, so keep these private
   flags: Flags,
   maybe_config_file: Option<ConfigFile>,
+  maybe_package_json: Option<PackageJson>,
   maybe_lockfile: Option<Arc<Mutex<Lockfile>>>,
   overrides: CliOptionOverrides,
+  npm_process_state: Arc<Mutex<Option<NpmProcessState>>>,
```

Comment:
Nitpick: might be more clear if this was a oncecell `Lazy`? Then we could return a reference to this I think in the method.